python/lsst/pipe/base/_observation_dimension_packer.py

```diff
@@ -0,0 +1,166 @@
+# This file is part of pipe_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("ObservationDimensionPacker", "ObservationDimensionPackerConfig", "observation_packer_registry")
+
+from typing import Any, cast
+
+from lsst.daf.butler import DataCoordinate, DimensionPacker
+from lsst.pex.config import Config, Field, makeRegistry
+
+observation_packer_registry = makeRegistry(
+    "Configurables that can pack visit+detector or exposure+detector data IDs into integers."
+)
+
+
+class ObservationDimensionPackerConfig(Config):
+    # Config fields are annotated as Any because support for betters
+    # annotations is broken on Fields with optional=True.
+    n_detectors: Any = Field(
+        "Number of detectors, or, more precisely, one greater than the "
+        "maximum detector ID, for this instrument. "
+        "Default (None) obtains this value from the instrument dimension record.",
```

Comment:
Tests is a lot of it; I also see it as a bit of an escape valve for unexpected issues with the values in the database, since changing those is immensely difficult, and sometimes changing the defaults in the code is similarly problematic (e.g. when working with a fixed release).  I've added a note about overrides being expected to be rare outside tests.

Reply:
I see the override note on n_observations, but not on n_detectors?

Reply:
I see the override note on n_observations, but not on n_detectors?